src/error/error.rs

```diff
@@ -5,43 +5,80 @@ use std::fmt;
 use http::status::StatusCode;
 
 /// Errors that can happen inside Tower Web.
+/// The object of this type is serializable into "Problem Detail" as defined in RFC7807.
+#[derive(Serialize)]
 pub struct Error {
-    kind: ErrorKind,
-}
-
-/// A list specifying the general categories of Tower Web errors.
-pub struct ErrorKind {
-    kind: KindPriv,
-}
-
-#[derive(Debug, Eq, PartialEq, Clone, Copy)]
-enum KindPriv {
-    BadRequest,
-    Unauthorized,
-    Fordidden,
-    NotFound,
-    UnprocessableEntity,
-    Internal,
+    #[serde(rename = "type")]
+    kind: String,
+    title: String,
+    // NOTE: type of this property might be changed in the future.
+    // Nevertheless, String type should be accepted.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    detail: Option<String>,
+    #[serde(skip)]
+    status: StatusCode,
+
+    // TODO: this property isn't used and should be removed
+    #[serde(skip)]
+    error_kind: ErrorKind,
 }
 
 // ===== impl Error =====
 
 impl Error {
+    // TODO: this function should return '&self.kind' property
+    // and its return value should be changed to '&str'
     /// Returns the corresponding `ErrorKind` for this error.
+    #[deprecated(note="return value of the function will be changed to &str")]
     pub fn kind(&self) -> &ErrorKind {
-        &self.kind
+        &self.error_kind
+    }
+
+    /// Creates a blank error object specifying only a status code without any details.
+    pub fn blank(status: StatusCode) -> Self {
```

Comment:
Could we rename this to `from_status` maybe?

Reply:
renamed

Reply:
I've replaced `from_status` with `impl From<http::StatusCode> for Error`. It seems even more convenient.